### PR TITLE
Adding sm csi driver and ascp installations

### PR DESCRIPTION
**Pull request by uchinda-sph**

### CODE REVIEW

**Review by niroz89 — APPROVED** [review]
LGTM, can help to add ASCP as well ? @lawliet89 can you help to take a look once.

**Comment by niroz89** [comment]
> LGTM, can help to add ASCP as well ? @lawliet89 can you help to take a look once.

Oops sorry, ascp also there.

**Comment by uchinda-sph** [comment]
hahaha yes it is, need some modifications as well im on it

File: iam.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module "secrets_manager_role" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 4.21.1"
+
+  role_name_prefix = coalesce(var.iam_role_name, "${var.cluster_name}-sm-csi-")
+  role_description = "EKS Cluster ${var.cluster_name} Secret Manager CSI Driver role"
+
+  attach_external_secrets_policy        = true
+  external_secrets_secrets_manager_arns = ["arn:aws:secretsmanager:${local.region}:${data.aws_caller_identity.current.account_id}:secret:*"]
```
Comment (pkailasu) [inline iam.tf]: @uchinda-sph can expose this arns as input so that we can provide access to specific secrets

**Review by lawliet89 — DISMISSED** [review]
(See new comments)

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  description = "Driver Resources"
+  type        = map(any)
+  default = {
+    requests = {
+      cpu    = "200m"
+      memory = "200Mi"
+    }
+    limits = {
+      cpu    = "50m"
+      memory = "100Mi"
```
Comment (lawliet89) [inline variables.tf]: Can limits be lower than requests?

Anyway, as a rule, let's always set memory request = memory limit to prevent OS level OOM.

Comment (uchinda-sph) [reply variables.tf]: Done

**Review by lawliet89 — APPROVED** [review]

**Pull request merged**